lib/awful/widget/launcher.lua

```diff
@@ -12,8 +16,10 @@ local button = require("awful.button")
 local launcher = { mt = {} }
 
 --- Create a button widget which will launch a command.
--- @param args Standard widget table arguments, plus image for the image path
--- and command for the command to run on click, or either menu to create menu.
+-- @tparam table args
+-- @tparam string args.image image path to an image
+-- @tparam string args.command command to run on click
+-- @tparam table args.menu table of items to create a menu
```

Comment:
From the function first line:
```lua
if not args.command and not args.menu then return end
```

So both arguments are optional but at least one is mandatory to the function run.

I'd use `-- @tparam[opt]` for both and stipulate in the function description that it require one of them for the function to run.

Reply:
Agreed. I'll fix this up later today